File: backend/src/main/java/io/blog/devlog/domain/sitemap/service/SitemapService.java
package io.blog.devlog.domain.sitemap.service;

import io.blog.devlog.domain.sitemap.dto.PostUrlDto;

public interface SitemapService {
    String generateSitemap();
    void generateSitemapXml(String sitemap);
    void addPostToSubSitemap(PostUrlDto postUrlDto);
    void deletePostFromSubSitemap(PostUrlDto postUrlDto);
    void createAllPostToSubSitemap();
}

File: backend/sitemap-service/src/main/kotlin/io/blog/sitemap/service/SitemapService.kt
package io.blog.sitemap.service

import io.blog.sitemap.dto.PostUrlDto
import java.io.File

interface SitemapService {
    fun generateSitemap() : String
    fun generateSitemapXml(sitemap: String)
    fun addPostToSubSitemap(postUrlDto: PostUrlDto)
    fun deletePostFromSubSitemap(postUrlDto: PostUrlDto)
    fun createAllPostToSubSitemap()
}